### PR TITLE
Add contributing guidelines

### LINKED ISSUE #158: Add a contributing guideline
This project is currently missing resources for to learn about the preferred workflows or the style guide. At least for me a contributing guideline is a strong factor I consider before contributing to a project. In my opinion it helps set the right expectations and cuts down the noise in PRs required to educate new contributors.

It would be nice to have at least a short guideline in the README or in a separate CONTRIBUTING file to establish some rules for this project.

### PR DESCRIPTION
**Pull request by awvwgk**
- adds a draft for contributing guidelines for fpm

I have written those guidelines from scratch some time ago, they appeared first in [this LGPL licensed project](https://github.com/grimme-lab/xtb/blob/55c5ea1d19799711426eb71fc15d93707ae7003f/CONTRIBUTING.md). I can contribute them under MIT license or any other fitting license for fpm.

*Feel free to push to this branch to adjust the guidelines to your liking.*

closes #158

### CODE REVIEW

**Comment by milancurcic** [comment]
Thank you @awvwgk. Most of it we can use but can be condensed without losing meaning. I'll take the next stab at it.

I prefer not requiring signed commits. If anything, we can say in the guidelines something like:

> By contributing to fpm, you certify that you own or are allowed to share the content of your contribution under the fpm license.

@certik what do you think?

**Comment by everythingfunctional** [comment]
I agree with @milancurcic that self-signed commits is not necessary. Everything else I think is fine.

**Comment by milancurcic** [comment]
I reused most of @awvwgk material and added concrete workflow steps. Let me know what you all think.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+## For New Contributors
+
+If you never created a pull request before, welcome :tada:.
+You can learn how from
+[this great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+
+Don't know where to start?
+You can start by looking through these
+[help-wanted issues](https://github.com/fortran-lang/fpm/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen).
```
Comment (LKedward) [inline CONTRIBUTING.md:124]: We're not using this label at the moment, so the link gives zero results.
Maybe we should just link to all open issues?

**Review by LKedward — APPROVED** [review]
This all looks good to me! Only left a few small comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (milancurcic) [reply CONTRIBUTING.md:126]: I agree, and if/when we start using help-wanted tag, people will see it from the list of all issues.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## For New Contributors
+
+If you have never created a pull request before, welcome :tada:.
+You can learn how from
+[this great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+
+Don't know where to start?
+You can start by looking through these
+[help-wanted issues](https://github.com/fortran-lang/fpm/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen).
```
Comment (milancurcic) [inline CONTRIBUTING.md:126]:
```suggestion
You can start by looking through the list of
￼[open issues](https://github.com/fortran-lang/fpm/issues).
```

**Review by milancurcic — APPROVED** [review]

**Comment by awvwgk** [comment]
@milancurcic I really like the guidelines you made out of this.

**Comment by milancurcic** [comment]
Unless there are objections, I will merge this tomorrow this time.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+## Reporting a bug
+
+A bug is a _demonstrable problem_ caused by the code in this repository.
+Good bug reports are extremely valuable to us—thank you!
+
+Before opening a bug report:
+
+1. Check if the issue has already been reported.
+   ([issues](https://github.com/fortran-lang/fpm/issues))
+2. Check if it's still an issue or it has been fixed?
```
Comment (certik) [inline CONTRIBUTING.md:38]:
```suggestion
2. Check if it is still an issue or it has been fixed?
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+4. What happens instead?
+
+This information will help the community diagnose the issue quickly and with
+minimal back-and-forth.
+
+## Suggesting a feature
+
+Before suggesting a new feature, take a moment to find out if it fits
+the scope of the project, or if it has already been discussed.
+It's up to you to provide a strong argument to convince the community of the
```
Comment (certik) [inline CONTRIBUTING.md:57]:
```suggestion
It is up to you to provide a strong argument to convince the community of the
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+benefits of this feature.
+Please provide as much detail and context as possible.
+If applicable, include a mocked-up snippet of what the output or behavior would
+look like with this feature implemented.
+"Crazy", out-of-the-box ideas are especially welcome.
+It's quite possible that we're not considering an unusually creative solution.
```
Comment (certik) [inline CONTRIBUTING.md:63]:
```suggestion
It's quite possible that we are not considering an unusually creative solution.
```

**Review by certik — APPROVED** [review]
I left some minor fixes.

I think this is fine. I personally, to be honest, rarely read such guidelines, and simply go ahead and contribute to a given GitHub or GitLab project in my best faith effort, trying to be helpful and not waste other people's time. But I think it does not hurt to have these guidelines, I think they are all common sense to me, so I am +1 to merge.

**Comment by milancurcic** [comment]
Thank you, @awvwgk!

**Pull request merged**